src/D2L.Security.OAuth2/Keys/JsonWebKey.cs

```diff
@@ -60,81 +63,153 @@ public abstract class JsonWebKey {
 		/// <param name="json">The json JWK</param>
 		/// <returns>A <see cref="JsonWebKey"/></returns>
 		public static JsonWebKey FromJson( string json ) {
+			if( !TryParseJsonWebKey( json, out var result, out var error, out var e, out _ ) ) {
+				throw new JsonWebKeyParseException( error, e );
+			}
+
+			return result;
+		}
+
+		public static bool TryParseJsonWebKey(
+			string json,
+			[NotNullWhen( true )]
+			out JsonWebKey? result,
+			[NotNullWhen( false )]
+			out string? error,
+			out Exception? exception,
+			out bool useEncKey
+		) {
 			Dictionary<string, object> data;
 			try {
 				data = JsonSerializer.Deserialize<Dictionary<string, object>>( json );
 			} catch ( JsonException e ) {
-				throw new JsonWebKeyParseException( "error deserializing JSON web key string", e );
+				result = null;
+				error = "error deserializing JSON web key string";
+				exception = e;
+				useEncKey = false;
+				return false;
 			}
 
 			if( data.ContainsKey( "use" ) && data[ "use" ] != null && data[ "use" ].ToString() != "sig" ) {
-				string msg = String.Format( "invalid 'use' value in JSON web key: {0}", data[ "use" ] );
-				throw new JsonWebKeyParseException( msg );
+				result = null;
+				error = "invalid 'use' value in JSON web key: " + data[ "use" ];
+				exception = null;
+				useEncKey = data[ "use" ].ToString() == "enc";
+				return false;
 			}

```

Comment:
I would maybe make this ignored in general (and not just "enc")? Perhaps a `ParseResult { Success, Error, Ignore }` vs `bool`?

It would ultimately be nice to be able to say "the key {kid} is not intended for singnatures" rather than "the key {kid} was not found" though.

Reply:
In other words rather than returning ignore from here I'd prefer if we just successfully parsed all valid JWKs and then moved the policy decisions elsewhere